File: src/main/kotlin/inflearn/java/_2_5_에라토스테네스_체.java
package inflearn.java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class _2_5_에라토스테네스_체 {
    public static int solution(int n) {

        int answer = 0;
        int[] toN = new int[n + 1];
        for(int i = 2; i < toN.length; i++) {
            if(toN[i] == 0) {
                answer++;

                for(int j = i; j < toN.length; j = j + i) {
                    toN[j] = 1;
                }
            }
        }

        return answer;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int input = Integer.parseInt(br.readLine());

        System.out.print(solution(input));
    }
}

File: src/main/kotlin/inflearn/kotlin/ch02/2_5 에라토스테네스 체.kt
package inflearn.kotlin.ch02

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

/**
 * 이펙티브 코틀린에서 다룬 방식이 더 깔끔함
 */
object `2_5 에라토스테네스 체` {
    fun solution(n: Int): Int {
        var answer = 0
        val ch = IntArray(n + 1)
        for (i in 2..n) {
            if (ch[i] == 0) {
                answer++
                var j = i
                while (j <= n) {
                    ch[j] = 1
                    j += i
                }
            }
        }
        return answer
    }

    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val br = BufferedReader(InputStreamReader(System.`in`))
        val input = br.readLine().toInt()
        print(solution(input))
    }
}